--- console/jfbterm/jfbterm-0.4.7/Makefile.am.old	2008-11-04 19:54:36.000000000 +0800
+++ console/jfbterm/jfbterm-0.4.7/Makefile.am	2008-11-04 20:01:28.000000000 +0800
@@ -28,5 +28,6 @@
 
 install-data-local:
 	$(INSTALL_DATA) $(srcdir)/$(JFBTERM_CONFIG_FILE) $(DESTDIR)$(sysconfdir)/$(JFBTERM_CONFIG_FILE)
-	$(INSTALL_DATA) $(srcdir)/fonts/*.pcf.gz $(DESTDIR)$(datadir)/fonts/jfbterm
+	mkdir -p $(DESTDIR)$(datadir)/fonts/jfbterm
+	$(INSTALL_DATA) $(srcdir)/fonts/*.pcf.gz $(DESTDIR)$(datadir)/fonts/jfbterm/
 
