--- jfbterm-0.4.7/Makefile.am.old	2009-02-26 15:54:18.000000000 +0800
+++ jfbterm-0.4.7/Makefile.am	2009-02-26 15:54:35.000000000 +0800
@@ -27,6 +27,6 @@
 	chmod ug+s $(DESTDIR)$(bindir)/jfbterm
 
 install-data-local:
-	$(INSTALL_DATA) $(srcdir)/$(JFBTERM_CONFIG_FILE) $(DESTDIR)$(sysconfdir)/$(JFBTERM_CONFIG_FILE)
-	$(INSTALL_DATA) $(srcdir)/fonts/*.pcf.gz $(DESTDIR)$(datadir)/fonts/jfbterm
+	$(INSTALL_DATA) -D $(srcdir)/$(JFBTERM_CONFIG_FILE) $(DESTDIR)$(sysconfdir)/$(JFBTERM_CONFIG_FILE)
+	$(INSTALL_DATA) -D $(srcdir)/fonts/*.pcf.gz $(DESTDIR)$(datadir)/fonts/jfbterm
 
--- jfbterm-0.4.7/Makefile.am.old	2009-02-26 16:02:19.000000000 +0800
+++ jfbterm-0.4.7/Makefile.am	2009-02-26 16:02:11.000000000 +0800
@@ -20,7 +20,7 @@
 
 JFBTERM_CONFIG_FILE = jfbterm.conf.sample
 
-EXTRA_DIST = $(JFBTERM_CONFIG_FILE)
+EXTRA_DIST = $(JFBTERM_CONFIG_FILE) fonts/vt100-graphics.pcf.gz
 
 install-exec-hook:
 	chown root:utmp $(DESTDIR)$(bindir)/jfbterm
--- jfbterm-0.4.7/Makefile.am.old	2009-02-26 16:06:54.000000000 +0800
+++ jfbterm-0.4.7/Makefile.am	2009-02-26 16:09:21.000000000 +0800
@@ -28,5 +28,10 @@
 
 install-data-local:
 	$(INSTALL_DATA) -D $(srcdir)/$(JFBTERM_CONFIG_FILE) $(DESTDIR)$(sysconfdir)/$(JFBTERM_CONFIG_FILE)
-	$(INSTALL_DATA) -D $(srcdir)/fonts/*.pcf.gz $(DESTDIR)$(datadir)/fonts/jfbterm
+	mkdir -p $(DESTDIR)$(datadir)/fonts/jfbterm
+	$(INSTALL_DATA) $(srcdir)/fonts/*.pcf.gz $(DESTDIR)$(datadir)/fonts/jfbterm
+
+uninstall-local:
+	rm -rvf $(DESTDIR)$(sysconfdir)/$(JFBTERM_CONFIG_FILE)
+	rm -rvf $(DESTDIR)$(datadir)/fonts/jfbterm

 
